### PR TITLE
Support rawRequest in callable context

### PR DESCRIPTION
**Pull request by laurenzlong**
### Description

Resolves https://github.com/firebase/firebase-functions-test/issues/41

### CODE REVIEW

File: spec/providers/https.spec.ts [hunk 1]
```diff
@@ -1,45 +1,72 @@
-import { expect } from 'chai';
+import {expect} from 'chai';
```
Comment (joehan) [inline spec/providers/https.spec.ts]: So, it looks like the rest of this repo has spaces around { objects } - for example, https://github.com/firebase/firebase-functions-test/blob/master/src/index.ts#L24. Firebase-tools also keeps this style: https://github.com/firebase/firebase-tools/blob/master/src/commands/apps-list.ts#L5
I think your linter might have broken this throughout the files in this PR - can you switch this back?

File: spec/providers/https.spec.ts [hunk 2]
```diff
@@ -1,45 +1,72 @@
...
+  });
 
-    it('should run the wrapped onCall function and return result', async () => {
-        const test = fft();
-        const result = await test.wrap(cfToUpperCaseOnCall)({message: 'lowercase'});
-        expect(result).to.deep.equal({msg: 'LOWERCASE', from: 'anonymous'});
-    });
+  it('should run the wrapped onCall function and return result', async () => {
+    const test = fft();
+    const result = await test.wrap(cfToUpperCaseOnCall)({message: 'lowercase'});
```
Comment (joehan) [inline spec/providers/https.spec.ts]: Nit - add a newline after this line and after you call test.wrap, to split up the arrange ,act, and assert steps of this test

File: spec/providers/https.spec.ts [hunk 3]
```diff
@@ -1,45 +1,72 @@
...
+  });
 
-    it('should accept auth params', async () => {
-        const test = fft();
-        const options = {auth: {uid: 'abc'}};
-        const result = await test.wrap(cfToUpperCaseOnCall)({message: 'lowercase'}, options);
-        expect(result).to.deep.equal({msg: 'LOWERCASE', from: 'abc'});
+  it('should accept auth params', async () => {
+    const test = fft();
+    const options = {auth: {uid: 'abc'}};
```
Comment (joehan) [inline spec/providers/https.spec.ts]: Nit - add a newline after this line and after you call test.wrap, to split up the arrange ,act, and assert steps of this test

File: spec/providers/https.spec.ts [hunk 4]
```diff
@@ -1,45 +1,72 @@
...
+      return {
+        session: {data: sessionData},
+      };
+    };
+    mockRequest.rawBody = Buffer.from('foobar');
+
+    const test = fft();
+    const options = {
+      rawRequest: mockRequest,
+    };
```
Comment (joehan) [inline spec/providers/https.spec.ts]: Nit - add a newline after this line and after you call test.wrap, to split up the arrange ,act, and assert steps of this test

File: spec/providers/https.spec.ts (hunk 1 above)
Comment (laurenzlong) [reply spec/providers/https.spec.ts]: Yes it did... I had the wrong VSCode settings.... thanks for catching.

**Comment by laurenzlong** [comment]
@joehan Fixed. PTAL

**Comment by laurenzlong** [comment]
Oh, actually I see you already approved. Thanks!

**Pull request merged**